_posts/news/2019-09-18-announcing-scalajs-0.6.29.md

```diff
@@ -0,0 +1,83 @@
+
+If upgrading from Scala.js 0.6.14 or earlier, make sure to read [the release notes of 0.6.15]({{ BASE_PATH }}/news/2017/03/21/announcing-scalajs-0.6.15/), which contain important migration information.
+
+As a minor release, 0.6.29 is backward binary compatible with previous releases in the 0.6.x series.
+Libraries compiled with earlier versions can be used with 0.6.29 without change.
+0.6.29 is not forward binary compatible with earlier releases: libraries compiled with 0.6.29 cannot be used by projects using 0.6.{0-28}.
+
+Please report any issues [on GitHub](https://github.com/scala-js/scala-js/issues).
+
+## Fix for a major bug in incremental compilation with sbt 1.x
+
+When using sbt-scalajs with sbt 1.x, there was a major bug with incremental compilation, causing linker errors ([#3419](https://github.com/scala-js/scala-js/issues/3419) / [#3596](https://github.com/scala-js/scala-js/issues/3596)).
+Scala.js 0.6.29 fixes this issue.
+
+## Major performance improvement notably for pattern matching
+
+Scala.js 0.6.29 features a major performance improvement to all instance tests of the form `x.isInstanceOf[C]` where `C` is a Scala `class`.
+This includes most cases of pattern matching.
+Applications that are pattern matching-intensive can experience speedups of up to 40%.
+
+## Dynamic imports with `js.import()`
+
+ECMAScript 2020 will standardize [dynamic imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Dynamic_Imports).
+Scala.js adds support for them with the `js.import[A <: js.Any]()` method, which returns a `js.Promise[A]`.
+The parameter `A` should be a JS trait describing the API of the module, and be given explicitly.
+Since `import` is a keyword in Scala, it must be called with backticks:
+
+{% highlight scala %}
+import scala.scalajs.js
+
+trait FooAPI extends js.Any {
+  def bar(x: Int): Int
+}
+
+val moduleName = "foo.js"
+val promise = js.`import`[FooAPI](moduleName)
+val future = promise.toFuture
+for (module <- future) {
+  println(module.bar(5))
+}
+{% endhighlight %}
+
+## Bug fixes
+
+Among others, the following bugs have been fixed in 0.6.29:
+
+* [#3419](https://github.com/scala-js/scala-js/issues/3419) / [#3596](https://github.com/scala-js/scala-js/issues/3596) Major bug in incremental compilation with sbt 1.x
+* [#3607](https://github.com/scala-js/scala-js/issues/3607) Calling `getClass.getSimpleName` on objects returns name without trailing `$`
+* [#3604](https://github.com/scala-js/scala-js/issues/3604) Implement `java.util.Collection.removeIf`
+* [#3605](https://github.com/scala-js/scala-js/issues/3605) `ArrayBuilder.generic.addAll` is broken on 2.13
+
+You can find the full list [on GitHub](https://github.com/scala-js/scala-js/issues?q=is%3Aissue+milestone%3Av0.6.29+is%3Aclosed).
```

Comment:
The following commits are entirely untracked but are bugfixes. This is not good.

https://github.com/scala-js/scala-js/commit/f0a0228ebf0f50234411157777e30fba22ce4741
https://github.com/scala-js/scala-js/commit/ed5d5f4334303b7db1e12eabc9f590e8506feae7
https://github.com/scala-js/scala-js/commit/5180897c2f9d8c5200a5b41bfd9892a20d6b755e
https://github.com/scala-js/scala-js/commit/cde963b2638120f07304ccad6e870dd6bb9f28ea
https://github.com/scala-js/scala-js/commit/52ac12ca00576cb6574b61de923d71de29c697c6

Reply:
I have created issues and closed them *a posteriori*, so that they can be tracked.